libqtile/backend/x11/core.py

```diff
@@ -165,10 +163,33 @@ def __init__(self, display_name: str = None) -> None:
         self._numlock_mask = xcbq.ModMasks.get(self.conn.get_modifier(numlock_code), 0)
         self._valid_mask = ~(self._numlock_mask | xcbq.ModMasks["lock"] | xcbq.AllButtonsMask)
 
+        self._mod_ks_to_mask = self._make_keysym_to_modmask_dict()
+
     @property
     def name(self):
         return "x11"
 
+    def _make_keysym_to_modmask_dict(self) -> Dict[int, int]:
+        """Make a dictionary mapping modifier keysyms to modifer masks.
+
+        This essentially gives the same information as `xmodmap -pm`, but
+        with keysyms mapping to modifiers instead of the reverse.
+        """
+        modmap_reply = self.conn.conn.core.GetModifierMapping().reply()
```

Comment:
I think the keysyms are likely to be constant across all users, so would it make sense to just add the dictionary directly, into `xcbq.py`, kinda like the `ModMasks` dictionary?

Reply:
Actually, they can be configured using `xmodmap`. I bet the intersection of qtile users and xmodmap users configuring their keymap is not empty... I'll still have a look though, maybe it turns out that `ModMasks` somehow contains the same information although I doubt it. It would be much easier if the dictionary was static....